### PR TITLE
Configurable Infil Templates

### PR DESCRIPTION
**Pull request by ArcaneData1**
- [x] infiltration templates set inside config

- [x] add alien hunters gear

resolves issues #31 and #32

### CODE REVIEW

File: CovertInfiltration/Config/XComInfiltration.ini [hunk 1]
```diff
@@ -62,4 +12,118 @@ RANKS_DETER[4] = 8
...
+; ALIEN HUNTERS ARMORS
++InfilModifiers=(Item="MediumAlienArmor",		HoursAdded=12, RiskReductionPercent=7)
++InfilModifiers=(Item="HeavyAlienArmor",		HoursAdded=15, RiskReductionPercent=7)
++InfilModifiers=(Item="HeavyAlienArmorMk2",		HoursAdded=15, RiskReductionPercent=7)
++InfilModifiers=(Item="LightAlienArmor",		HoursAdded=9, RiskReductionPercent=7)
++InfilModifiers=(Item="LightAlienArmorMk2",		HoursAdded=9, RiskReductionPercent=7)
+; DISGUISES
++InfilModifiers=(Item="CivilianDisguise",		HoursAdded=3, RiskReductionPercent=1, MultiplierCategory="weapon", InfilMultiplier=2)
++InfilModifiers=(Item="AdventDisguise",			HoursAdded=3, RiskReductionPercent=1, MultiplierCategory="weapon", InfilMultiplier=1.5)
++InfilModifiers=(Item="HolographicDisguise",	HoursAdded=3, RiskReductionPercent=1, MultiplierCategory="weapon", InfilMultiplier=1)
```
Comment (Xymanek) [inline CovertInfiltration/Config/XComInfiltration.ini]: What's the point of `InfilMultiplier=1`?

File: CovertInfiltration/Config/XComInfiltration.ini [hunk 2]
```diff
@@ -62,4 +12,118 @@ RANKS_DETER[4] = 8
...
++InfilModifiers=(Item="WristBlade_CV",	HoursAdded=3)
++InfilModifiers=(Item="WristBlade_MG",	HoursAdded=3)
++InfilModifiers=(Item="WristBlade_BM",	HoursAdded=3)
+
++InfilModifiers=(Item="PsiAmp_CV", HoursAdded=-6)
++InfilModifiers=(Item="PsiAmp_MG", HoursAdded=-6)
++InfilModifiers=(Item="PsiAmp_BM", HoursAdded=-6)
+
++InfilModifiers=(Item="HoloProjector",	HoursAdded=0, RiskReductionPercent=0, MultiplierCategory="weapon",	InfilMultiplier=0.5)
++InfilModifiers=(Item="AdaptiveCamo",	HoursAdded=0, RiskReductionPercent=0, MultiplierCategory="armor",	InfilMultiplier=0.5)
```
Comment (Xymanek) [inline CovertInfiltration/Config/XComInfiltration.ini]: Get rid of `HoursAdded=0, RiskReductionPercent=0`, that's the default values anyway

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplate.uc [hunk 3]
```diff
@@ -10,20 +10,17 @@
 class X2InfiltrationModTemplate extends X2DataTemplate;
 
 // Number of hours this item adds to the Covert Action clock
-var int InfilModifier;
+var int HoursAdded;
 // When equipping this item, items with an ItemCat matching MultCategory
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplate.uc]: Add a blank line before full line comments (this and `// How much [...]` later). Again, for readability

**Comment by Xymanek** [comment]
Also, I think it makes sense overriding `ValidateTemplate` in `X2InfiltrationModTemplate` to check that 

* `MultCategory != ''` when `InfilMultiplier != 1`
* `InfilMultiplier != 1` when `MultCategory != ''`

I separated those 2 into separate cases so that the `strError` is more meaningful

**Comment by ArcaneData1** [comment]
I disagree about removing the spaces from the config lines. Keeping the columns aligned lets someone view all the values at a glance, making it easier to balance the numbers. Its also more readable to users (including non-programmers) who want to modify or add entries to the config themselves.

File: CovertInfiltration/Config/XComInfiltration.ini (hunk 2 above)
Comment (ArcaneData1) [reply CovertInfiltration/Config/XComInfiltration.ini]: Again, I think the current way is clearer to a non-programmer.

**Comment by Xymanek** [comment]
After opening the file in text editors I can see your point. It still looks messed up on github but whatever, it's probably worth it.
Also, you make a good point about this being non-programmer friendly. I didn't consider that

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplate.uc [hunk 4]
```diff
@@ -9,19 +9,29 @@
...
+		strError = "CI: Error in templates, InfilMultiplier has been set but MultiplierCategory is blank.";
+		return false;
+	}
+	else if (MultCategory != '' && InfilMultiplier == 1)
+	{
+		strError = "CI: Error in templates, MultiplierCategory has been set but InfilMultiplier remains default.";
+		return false;
+	}
+
+	return super.ValidateTemplate(strError);
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2InfiltrationModTemplate.uc]: Change this to `return true` and I think this PR is ready to be merged

**Pull request merged**